cmd/bblfshd/main.go

```diff
@@ -197,9 +197,48 @@ func listenUser(d *daemon.Daemon) {
 	}
 }
 
+func unwrap(err error) error {
+	switch err := err.(type) {
+	case *net.OpError:
+		return unwrap(err.Err)
+	case *os.SyscallError:
+		return unwrap(err.Err)
+	}
+	return err
+}
+
+func isErrno(err error, eno syscall.Errno) bool {
+	err = unwrap(err)
+	if e, ok := err.(syscall.Errno); ok && e == eno {
+		return true
+	}
+	return false
+}
+
+func isAddrInUse(err error) bool {
+	return isErrno(err, syscall.EADDRINUSE)
+}
+
+func isConnRefused(err error) bool {
+	return isErrno(err, syscall.ECONNREFUSED)
+}
+
 func listenControl(d *daemon.Daemon) {
 	var err error
 	ctlListener, err = net.Listen(*ctl.network, *ctl.address)
+	if isAddrInUse(err) && *ctl.network == "unix" {
```

Comment:
Do we need to do all this? This socket is inside the container, not on the host, right? So in principle we should not have multiple bblfshd instances trying to use it. If that's correct, we can blindly `os.Remove` the socket path before we attempt to bind, and trust that we're either a restart (if it exists) or the first (if not).

Reply:
You are right, it's _usually_ in the container. But we also got multiple requests for running it without the container, that is possible right now. So I think it's better to be on a safe side and assume some might run two of those by accident.

Reply:
Even if it's running outside the container, though, we don't support running multiple instances on the same host (that's implicit in the fact that we use the same address).

So I don't think this should be necessary even outside the container. If we're starting up and we have a Unix-domain socket, I think we can just remove it before binding. (I think that's what Stevens recommends in Unix Network Programming, and it seems to be fairly common)

Reply:
Sounds reasonable. Will change it.

Reply:
Done

Reply:
I belive this comment below is relevant to this thread https://github.com/bblfsh/bblfshd/commit/f629bc9f5ea1ef7890b57223d78281a962f128ab#r33405791